### PR TITLE
🌱 hack/observability: improve Grafana dashboards

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/observability/grafana/dashboards/cluster-api.json [hunk 1]
```diff
@@ -557,514 +557,613 @@
       "type": "stat"
     },
     {
-      "collapsed": false,
+      "collapsed": true,
```
Comment (sbueringer) [inline hack/observability/grafana/dashboards/cluster-api.json:560]: Just changed it everywhere that all rows are collapsed per default. This just avoids a bit of unnecessary metrics loading when the dashboard is opened

File: hack/observability/grafana/dashboards/cluster-api.json [hunk 2]
```diff
@@ -557,514 +557,613 @@
...
+              "expr": "avg((capi_machine_status_condition_last_transition_time{status=\"True\"}>= time() - 60*10) - on(name) group_left () (capi_machine_created * on (name) group_right avg(capi_machine_info{control_plane_name!=\"\"}) by (name))) by (type)",
+              "legendFormat": "{{type}}",
+              "range": true,
+              "refId": "A"
+            }
           ],
-          "displayMode": "table",
-          "placement": "right",
-          "showLegend": true
+          "title": "Control Plane Machine time until condition true since creationTimestamp",
```
Comment (sbueringer) [inline hack/observability/grafana/dashboards/cluster-api.json:860]: I changed this so now we can see this separate for control plane and worker machines. Before it was just one. Separating it helps to make more sense out of the data as the data is fundamental different for cp and worker machines

File: hack/observability/grafana/dashboards/controller-runtime.json [hunk 3]
```diff
@@ -2681,7 +2826,7 @@
                 "uid": "PBFA97CFB590B2093"
               },
               "editorMode": "code",
-              "expr": "histogram_quantile(0.50, sum(rate(workqueue_queue_duration_seconds_bucket{pod=~\"$Pod\",name=~\"$Controller\"}[5m])) by (pod, name, le))",
```
Comment (sbueringer) [inline hack/observability/grafana/dashboards/controller-runtime.json:2684]: Changed the order of panels. Now the first one is work duration and the second one queue duration which seems more intuitive (I'm always looking for work duration)

**Review by sbueringer — COMMENTED** [review]
Added some explanations, The diff view in GitHub is horrible. I would recommend e.g. Intellij. Itl ooks a lot cleaner there. Feel free to ask if some changes are not clear.

File: test/e2e/data/infrastructure-inmemory/main/cluster-template.yaml
[the remark this answers is not in the document]
Comment (sbueringer) [reply test/e2e/data/infrastructure-inmemory/main/cluster-template.yaml:1]: Thx, dropped. Time that we merge the e2e test so I don't have to be careful with the file that should be on .gitignore :)

**Comment by stmcginnis** [comment]
/lgtm

But I can't fully test. Can you add some screenshots of the updated results to help see the new state?

File: hack/observability/grafana/dashboards/cluster-api.json [hunk 4]
```diff
@@ -557,514 +557,1202 @@
...
+            }
+          },
+          "targets": [
+            {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "PBFA97CFB590B2093"
+              },
+              "editorMode": "code",
+              "expr": "avg(min_over_time(capi_cluster_status_condition_last_transition_time{status=\"True\"}[168h]) - on(name) group_left () capi_cluster_created) by (type)",
```
Comment (sbueringer) [inline hack/observability/grafana/dashboards/cluster-api.json:658]: Made some further improvments to this dashboard:
* now using min_over_time => we get the min transition time within the last week (i.e. the first time the condition flipped to true)
* I already had panels which only showed the what transitioned to true in the last 5m, now additionally we have panels for all currently existing objects

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve
/hold

Feel free to unhold if you're happy with the current state - looks great to me!

**Review by killianmuldoon — COMMENTED** [review]
/area devtools

**Pull request merged**